File: Data Structures & Algorithms/top-k-elements-in-list/submission-9.cpp
#include <algorithm>
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> result;
        std::map<int, std::vector<int>> counter;
        std::unordered_set<int> seen;

        for (int &num: nums) {
            if (!seen.count(num)) {
                counter[std::count(nums.begin(), nums.end(), num)].push_back(num);
                seen.insert(num);
            }
            
        }
    
        while (k) {
            for (auto cit = counter.crbegin(); cit != counter.crend(); ++cit){
                for (const int &ele: cit->second) {
                    if (k == 0) return result;
                    else {
                        result.push_back(ele);
                        k--;
                    }
                }
            }
        }
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-9.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:

        prefix = nums[0]

        result = [1] * len(nums)

        for i in range(1, len(nums)):
            result[i] = prefix
            prefix *= nums[i]
        
        sufix = nums[-1]

        for i in range(len(nums) - 2, -1, -1):
            result[i] *= sufix
            sufix *= nums[i]

        return result
        